cmd/check_container.go

```diff
@@ -152,111 +152,117 @@ var checkContainerCmd = &cobra.Command{
 			return err
 		}
 
-		// submitting results to pxysis if submit flag is set
+		// assemble artifacts and submit results to pyxis if user provided the submit flag.
 		if submit {
 			log.Info("preparing results that will be submitted to Red Hat")
-			log.Tracef("CertProject: %+v", certProject)
+			log.Tracef("CertProject: %+v", certProject) // TODO() This depends on Pyxis calls made outside of this block.
 
-			testResultsJsonFile, err := os.Open(path.Join(artifacts.Path(), certification.DefaultTestResultsFilename))
+			// The engine writes the certified image config to disk in a Pyxis-specific format.
+			// Include that data in our submission.
+			certImageFilepath := path.Join(artifacts.Path(), certification.DefaultCertImageFilename)
+			log.Tracef("Loading certImage data from path %s", certImageFilepath)
+			certImageBytes, err := os.ReadFile(certImageFilepath)
 			if err != nil {
-				return err
-			}
-			defer testResultsJsonFile.Close()
-
-			testResultsBytes, err := io.ReadAll(testResultsJsonFile)
-			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: unable to read file from disk to include in submission: %s: %s",
+					errors.ErrSubmittingToPyxis,
+					certImageFilepath,
+					err,
+				)
 			}
 
-			testResults := new(pyxis.TestResults)
-			err = json.Unmarshal(testResultsBytes, &testResults)
+			var certImage pyxis.CertImage
+			err = json.Unmarshal(certImageBytes, &certImage)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"certImage",
+					err,
+				)
 			}
 
-			certImageJsonFile, err := os.Open(path.Join(artifacts.Path(), certification.DefaultCertImageFilename))
+			// The certification engine also writes the rpmManifest for images not based on scratch.
+			// Include that data in our submission.
+			rpmManifestFilepath := path.Join(artifacts.Path(), certification.DefaultRPMManifestFilename)
+			log.Tracef("Loading RPM manifest data from path %s", rpmManifestFilepath)
+			rpmManifestBytes, err := os.ReadFile(rpmManifestFilepath)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: unable to read file from disk to include in submission: %s: %s",
+					errors.ErrSubmittingToPyxis,
+					rpmManifestFilepath,
+					err,
+				)
 			}
-			defer certImageJsonFile.Close()
 
-			certImageBytes, err := io.ReadAll(certImageJsonFile)
+			var rpmManifest pyxis.RPMManifest
+			err = json.Unmarshal(rpmManifestBytes, &rpmManifest)
 			if err != nil {
-				return err
+				return fmt.Errorf(
+					"%w: data for the %s appears to be malformed: %s",
+					errors.ErrSubmittingToPyxis,
+					"rpm manifest",
+					err,
+				)
 			}
 
-			certImage := new(pyxis.CertImage)
-			err = json.Unmarshal(certImageBytes, &certImage)
+			// Include Preflight's test results in our submission. pyxis.TestResults embeds them.
+			var testResults pyxis.TestResults
+			err = json.Unmarshal(formattedResults, &testResults)
```

Comment:
Since formattedResults isn't really the struct that testResults is built on, and does not have an `json` values, does this actually work as expected? ie has the payload sent to pyxis been validated?

Reply:
@acornett21 If I understood correctly, the file we read from disk was results.json. We write formattedResults to that file, no? 

In theory, this is not a very safe/reliable operation either way because, while we don't really expose it, the code supports other formatters for the UserResponse. Either way, I believe the data we were reading in using the previous implementation is the same.

Correct me if I'm wrong!

Alternatively, maybe we find an appropriate way to expose [formatters.getResponse](https://github.com/redhat-openshift-ecosystem/openshift-preflight/blob/main/certification/formatters/util.go#L10)'s behavior?

Reply:
Looking more, I was missing that this uses `genericJSONFormatter` method which in turn uses the `formatters.UserResponse` which is in `pyxis.TestResults`, so under the covers they are the same, just getting there is different and hard to see just from this one change.